packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -614,7 +615,15 @@ export class ConnectionManager implements IConnectionManager {
 					// We skip this delay if we're confident we're offline, because we probably just need to wait to come back online.
 					await new Promise<void>((resolve) => {
 						setTimeout(resolve, delayMs);
-						delayMs = Math.min(delayMs * 2, MaxReconnectDelayInMs);
+						// If the error has some status code, it means we probably reached to endpoint and got some error. In that case
+						// set the max time to be 8s as we want to retry at faster pace. Otherwise, wait for MaxReconnectDelayInMs at max.
+						delayMs = Math.min(
+							delayMs * 2,
+							isFluidError(origError) &&
+								origError.getTelemetryProperties().statusCode !== undefined
```

Comment:
Relying on presence of status code here is a bit dangerous - it's not well-defined interface and it's not given that some driver would use same structure (Do we use statusCode property in FRS? That failures on the socket would have status code, though it might be minor, as any error on the socket should fallback to joinSession / discovery - in both cases these are REST calls). While it's only slightly better, I'd look for ways to use errorType for that kind of check, as it's more uniform (and expected through layers) structure of errors (to have it, and we clearly tell hosts a list of supported errorTypes).

Looking at fetchHelper() though, that would be OdspErrorType.fetchTimeout, DriverErrorType.offlineError, DriverErrorType.fetchFailure (and maybe DriverErrorType.incorrectServerResponse should be included) - too many different error types, and we would risk missing one if/when one more is added.
So, I'm a little bit torn here on the best course of action.

For that reason, I'd rather add a new property to clearly differentiate - fetch that returned some response, and fetch that did not. We already have odspError: true, storageServiceError: true for cases where ODSP has response. Can we use this kind of flow (maybe add new property? Maybe use storageServiceError) to indicate that we indeed got some response from service? And use it uniformly? If so, can we add it to IDriverErrorBase definition to create slightly stronger contract? This might require some code reshuffle, as (for example) some of the properties are set by throwOdspNetworkError() & errorObjectFromSocketError(), not by createOdspNetworkError() (called from both places), which seems wrong.